### PR TITLE
go-sdk: Pipeline notification changes

### PR DESCRIPTION
**Pull request by blinktag**
* Bump protos to v0.1.21
* Update `NotifyRequest` with new fields
* Fix README.md example

### CODE REVIEW

File: sdks/go/go_sdk.go
Comment (dselans) [inline sdks/go/go_sdk.go:881]: Agreed - this is getting really long. I have to work on the SDK next, I'll do this.

File: sdks/go/server/client.go
Comment (dselans) [inline sdks/go/server/client.go:37]: Really good that we can specify concrete types.

**Review by dselans — APPROVED** [review]
Let's get this merged!

**Review by dselans — CHANGES_REQUESTED** [review]
Undo the README.md changes - I had to update it a bit ago because of the SDK migration.

**Comment by dselans** [comment]
I rebased on main + removed the README updates.

**Review by dselans — APPROVED** [review]

File: sdks/go/README.md
Comment (blinktag) [reply sdks/go/README.md]: Sweet! Done

**Comment by blinktag** [comment]
Waiting for frontend changes to land before merging

**Pull request merged**
